drivers/internal/userdata/userdata.go

```diff
@@ -28,8 +30,19 @@ func Parse(text string) *template.Template {
 	)
 }
 
-// T is the default userdata template.
-var T = Parse(`#cloud-config
+// DetectUserdata returns the docker provisioning userdata script based on the OS string
+func DetectUserdata(os string) (*template.Template, error) {
+	switch {
```

Comment:
lets move this to a separate pull request and keep this PR focused just on the packet implementation. This will likely always fail for amazon, since the image is formatted ami-80ca47e6 and does not contain amazon.